src/Deserialization/Extensions/BasicBlockRefExtensions.cs

```diff
@@ -0,0 +1,21 @@
+﻿using System.Collections.Generic;
+using LLVMSharp.Interop;
+
+namespace Symbolica.Deserialization.Extensions
+{
+    internal static class BasicBlockRefExtensions
+    {
+        public static IEnumerable<LLVMValueRef> GetInstructions(this LLVMBasicBlockRef self)
+        {
+            var instruction = self.FirstInstruction;
+
+            while (instruction != self.LastInstruction)
+            {
+                yield return instruction;
+                instruction = instruction.NextInstruction;
+            }
+
+            yield return instruction;
```

Comment:
These "iterators" are really weird. They don't work like their C++ equivalents where "end" is actually "one past the end", so you can cleanly loop them. Clearly, these cannot support the case when there are zero Functions, Globals, or BasicBlocks, but I don't believe this degenerate case can ever actually happen, so simply yielding one "extra" should be fine.